.github/workflows/codeql-analysis.yml

```diff
@@ -40,7 +40,7 @@ jobs:
       uses: github/codeql-action/init@main
       # Override language selection by uncommenting this and choosing your languages
       with:
-        languages: csharp
+        languages: csharp, cpp
```

Comment:
Can't comment on the Bazel changes, but consider matrixing the analysis over languages (e.g. see the template here https://github.com/actions/starter-workflows/blob/main/code-scanning/codeql.yml).  This is what we recommend to customers as it means the analyses can run in parallel, and if one fails it doesn't disrupt the other.

Reply:
I think when no category is specified we fall back to the workflow name, job name, and matrix parameters e.g. here `.github/workflows/codeql-analysis.yml:CodeQL-Build`.  So actually matrixing the build would create new histories when it comes to alert matching.  I think that's probably a price we're willing to pay, but it's less of a clear cut win.

Reply:
Ah my mistake, wrong way round. Specifying the category when we do that is probably worthwhile then.

Reply:
How do you feel about creating a second workflow file for Swift analysis? The build commands are completely different and independent.

Reply:
hmm, seems like a good idea to me! Then it shouldn't disrupt the current analysis state in any way